addons/bus/static/src/js/web_client_bus.js

```diff
@@ -0,0 +1,92 @@
+odoo.define('bus.WebClient', function (require) {
+"use strict";
+
+const core = require('web.core');
+const WebClient = require('web.WebClient');
+
+const _t = core._t;
+
+WebClient.include({
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    /**
+     * Detects the presence of assets in DOM's HEAD
+     *
+     * @override
+     */
+    async start() {
+        this._assetsChangedNotificationId = null;
+        this._assets = {};
+        this.call('bus_service', 'onNotification', this, this._onNotification);
+        await this._super.apply(this, arguments);
+    },
+    /**
+     * Assign handler to bus notification
+     *
+     * @override
+     */
+    show_application() {
+        const showed = this._super.apply(this, arguments);
+        document.querySelectorAll('*[data-asset-xmlid]').forEach(el => {
+            this._assets[el.getAttribute('data-asset-xmlid')] = el.getAttribute('data-asset-version');
+        });
+        this.call('bus_service', 'startPolling');
+        return showed;
+    },
+
+    //----------------------------------------------------------------------
+    // Private
+    //----------------------------------------------------------------------
+
+    /**
+     * Displays one notification on user's screen when assets have changed
+     *
+     * @private
+     */
+    _displayBundleChangedNotification() {
+        if (!this._assetsChangedNotificationId) {
+            this._assetsChangedNotificationId = this.call('notification', 'notify', {
+                title: _t('Update available'),
+                message: _t('Refresh your browser to take advantage of the latest update of Odoo in your browser'),
+                sticky: true,
+                onClose: () => {
+                    this._assetsChangedNotificationId = null;
+                },
+                buttons: [{
+                    text: _t('Refresh'),
+                    primary: true,
+                    click: () => {
+                        window.location.reload(true);
+                    }
+                }],
+             });
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Reacts to bus's notification
+     *
+     * @private
+     * @param {Array} notifications: list of received notifications
+     */
+    _onNotification(notifications) {
+        for (const notif of notifications) {
+            if (notif[0][1] === 'bundle_changed') {
```

Comment:
What follows doesn't apply specifically to your commit, it is a general remark. This whole notification mechanism is badly designed (in JS at least). The service should process the notification, and dispatch them by types (other components could register to listen on specific types for instance). This would allow to stop looping on every notifications that occur, whatever their type, when you want to listen to a specific one (e.g. bundle_changed). This is something we could/should improve in the future, imho.